### ISSUE TITLE
Finals question format

### ISSUE THREAD

**Issue by zandratay** [nus-cs2030/2223-s1#322]
Hello, i would like to clarify -- the review videos posted on LumiNus web lectures are all pertaining to videos that are purely coding. I would like to ask if our final is purely coding too, like what was shown in the review videos, or are there theory questions too? 

in addition, is jshell allowed during the exam?

**Comment by Grauda123** [comment]
the finals would be similar to the questions in the review videos, also there wont be any MCQs too

> in addition, is jshell allowed during the exam?
Yes we can use jshell during the exam

**Comment by Wenbo-J** [comment]
the format will be similar to the review videos, it's pretty much a PA3 hhahah

**Comment by mohddanesh** [comment]
like what others say, it's just like another PA, but it's done on ur own pc and you don't get to modify ur code after the exam.

**Comment by ZYN2122** [comment]
We will be able to use basically all the software (ubuntu etc) as well?

**Comment by nguyenhpnguyen** [comment]
@ZYN2122 yes we use our laptops so you can edit the codes with your usually used software

**Comment by liewyungjun** [comment]
prof said u can basically use anything with the restriction being no internet except for codecrunch and java api

**Comment by limyeechern** [comment]
It's like another PA but we can use our own IDE so its a chance to discard vim and use vscode or intellij
